### PR TITLE
Feature/2872 97 update dependencies

### LINKED ISSUE #97: Update to latest dependencies
#### Describe the desired behavior
Use the most recent version of dependencies

#### Describe the current behavior
Using some outdated dependencies

#### Is this request related to a current issue?
No

#### Additional Context

Be particularly careful with updating these as there are compatibility consideration to account for. Test this VERY thoroughly as you go along to make sure that there are no regressions or other unintended consequences.

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes #97 

#### Changes proposed in this Pull Request:
- Updated package json

#### To Test:
- yarn start:example

### CODE REVIEW

**Comment by bkarambe** [comment]
react-router-dom will be updated https://github.com/brightlayer-ui/react-workflows/issues/102

**Comment by ektasawant** [comment]
Changes looks good. No error in consoles 👍 .
Login page
Screenshot 2022-03-31 at 11 40 43 AM

Home page
Screenshot 2022-03-31 at 11 40 59 AM

**Review by ektasawant — APPROVED** [review]

File: login-workflow/src/screens/SelfRegistrationPager.tsx [hunk 1]
```diff
@@ -219,6 +219,7 @@ export const SelfRegistrationPager: React.FC = () => {
                 <AcceptEula
                     eulaAccepted={eulaAccepted}
                     onEulaChanged={setEulaAccepted}
+                    // eslint-disable-next-line @typescript-eslint/no-misused-promises
```
Comment (joebochill) [inline login-workflow/src/screens/SelfRegistrationPager.tsx:222]: Why is this a misused promise?

File: login-workflow/package.json [hunk 2]
```diff
@@ -26,7 +26,7 @@
     "prettier": "@brightlayer-ui/prettier-config",
     "dependencies": {
         "i18next-browser-languagedetector": "^6.1.0",
-        "@brightlayer-ui/react-auth-shared": "^3.6.0-beta.1",
+        "@brightlayer-ui/react-auth-shared": "^3.6.1-beta.0",
```
[the remark this answers is not in the document]
Comment (bkarambe) [reply login-workflow/package.json:29]: No there is no WIP for this. We can publish.

File: login-workflow/src/screens/SelfRegistrationPager.tsx (hunk 1 above)
Comment (bkarambe) [reply login-workflow/src/screens/SelfRegistrationPager.tsx:222]: **loadEula** accepts void return and we are returning a Promise. That's why I disabled eslint here. If we change the return type of loadEula, we are not using it as a promise.

File: login-workflow/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (joebochill) [reply login-workflow/package.json:29]: Cool, let's add that to our list of packages that we need to publish

File: login-workflow/src/screens/SelfRegistrationPager.tsx (hunk 1 above)
Comment (joebochill) [reply login-workflow/src/screens/SelfRegistrationPager.tsx:222]: Do you think it makes more sense to put the eslint disable here or to update the type, i.e.:
image

File: login-workflow/babel.config.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
 module.exports = {
     presets: ['@babel/preset-env', '@babel/preset-react', '@babel/preset-typescript'],
+    plugins: ['@babel/transform-runtime'],
```
[the remark this answers is not in the document]
Comment (bkarambe) [reply login-workflow/babel.config.js:3]: I updated babel to v7, babel-polyfill has been deprecated in that, which will let us use custom regenerator runtime will will need this plugin.
Here's the screenshot, if you run **yarn test**. 
Screenshot 2022-03-31 at 6 31 24 PM

File: login-workflow/package.json [hunk 4]
```diff
@@ -39,8 +39,8 @@
         "@brightlayer-ui/react-components": "^6.0.0-alpha.3",
         "date-fns": "^2.14.0",
         "i18next": "^19.3.4 || ^20.0.0",
-        "react": "^16.13.1 || ^17.0.0",
-        "react-dom": "^16.8.6 || ^17.0.0",
+        "react": "^16.13.1 || ^17.0.0 || ^18.0.0",
+        "react-dom": "^16.8.6 || ^17.0.0 || ^18.0.0",
```
[the remark this answers is not in the document]
Comment (bkarambe) [reply login-workflow/package.json:43]: Looks like React 18 does not support ReactDOM.render. Reverting this back to use React 17

File: login-workflow/src/screens/SelfRegistrationPager.tsx (hunk 1 above)
Comment (bkarambe) [reply login-workflow/src/screens/SelfRegistrationPager.tsx]: Yes, I tried updating the loadEula return type to ()=>Promise<void> | void; but forgot to use void in call. Changes have been updated.

**Review by joebochill — APPROVED** [review]

**Pull request merged**